zboxcore/sdk/copyworker.go

```diff
@@ -160,15 +162,30 @@ func (req *CopyRequest) ProcessCopy() error {
 			defer wg.Done()
 			refEntity, err := req.copyBlobberObject(req.blobbers[blobberIdx], blobberIdx)
 			if err != nil {
+				select {
+					case wgErrors <- err:
+					default:
+				}
 				l.Logger.Error(err.Error())
 				return
 			}
 			objectTreeRefs[blobberIdx] = refEntity
 		}(int(pos))
 	}
-
 	wg.Wait()
 
+	var recentErr *error
+	select {
+		case err := <-wgErrors:
```

Comment:
line 153 + line 166 + line 175 + line 179 might make deadlock. @Manish-210  could you add an unit test to verify it?

Reply:
@cnlangzi 
The unit test here will have 2 blobbers giving correct responses from the 4 blobbers and it passes this test.

https://github.com/0chain/gosdk/blob/fix_incorrect_move_file_error_message/zboxcore/sdk/copyworker_test.go#L420

Is there any other unit test to be added?

Reply:
@cnlangzi it would not create deadlock, the `select...default` will unblock it.

Reply:
yes, ideally we should use buffered channel, the following changes that @Manish-210 has done should be fine as well.